app/src/main/java/com/example/chessclockk/screens/main/MainScreen.kt

```diff
@@ -0,0 +1,101 @@
+package com.example.chessclockk.screens.main
+
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Pause
+import androidx.compose.material.icons.filled.PlayArrow
+import androidx.compose.material.icons.filled.Refresh
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import com.example.chessclockk.views.clock.ClockState
+import com.example.chessclockk.views.playpause.PlayPauseState
+import com.example.chessclockk.views.restart.RestartState
+import com.example.chessclockk.vm.GameState
+import com.example.chessclockk.vm.IMainActivityVM.MainScreenState
+import com.example.chessclockk.vm.MainActivityVM
+
+@Composable
+
+fun MainScreen(
+    viewModel: MainActivityVM,
+    modifier: Modifier
+) {
```

Comment:
Przy argumentach composable trzeba bardzo uważać. Tutaj akurat lepiej podać model i metody w formie lambd.
Co do modelu to to też jest tricky, bo na przykład zwykłe "List" jest uważane jako mutable i trzeba używać "ImmutableList". Czasami nie da rady tak dopieścić modelu żeby był immutable i wtedy dajemy tam adnotację @Immutable żeby rekompozycja była sprawniejsza.
Być moze są już sprawniejsze metody sprawdzania rekompozycji, ale ja używałem tej https://developer.android.com/develop/ui/compose/performance/stability/diagnose#compose-compiler
Tutaj generujesz raporty i w ich treści możesz znaleźć jakie statusy mają metody i ich argumenty.